x/epochs/go.mod

```diff
@@ -0,0 +1,148 @@
+	github.com/matttproud/golang_protobuf_extensions v1.0.2-0.20181231171920-c182affec369 // indirect
+	github.com/mimoo/StrobeGo v0.0.0-20181016162300-f8f6d4d2b643 // indirect
+	github.com/minio/highwayhash v1.0.2 // indirect
+	github.com/mitchellh/mapstructure v1.5.0 // indirect
+	github.com/mtibben/percent v0.2.1 // indirect
+	github.com/pelletier/go-toml v1.9.5 // indirect
+	github.com/pelletier/go-toml/v2 v2.0.2 // indirect
+	github.com/petermattis/goid v0.0.0-20180202154549-b0b1615b78e5 // indirect
+	github.com/pmezard/go-difflib v1.0.0 // indirect
+	github.com/prometheus/client_golang v1.13.0 // indirect
+	github.com/prometheus/client_model v0.2.0 // indirect
+	github.com/prometheus/common v0.37.0 // indirect
+	github.com/prometheus/procfs v0.8.0 // indirect
+	github.com/rcrowley/go-metrics v0.0.0-20200313005456-10cdbea86bc0 // indirect
+	github.com/regen-network/cosmos-proto v0.3.1 // indirect
+	github.com/rs/cors v1.8.2 // indirect
+	github.com/rs/zerolog v1.27.0 // indirect
+	github.com/sasha-s/go-deadlock v0.2.1-0.20190427202633-1595213edefa // indirect
+	github.com/spf13/afero v1.8.2 // indirect
+	github.com/spf13/jwalterweatherman v1.1.0 // indirect
+	github.com/subosito/gotenv v1.4.0 // indirect
+	github.com/syndtr/goleveldb v1.0.1-0.20200815110645-5c35d600f0ca // indirect
+	github.com/tendermint/btcd v0.1.1 // indirect
+	github.com/tendermint/crypto v0.0.0-20191022145703-50d29ede1e15 // indirect
+	github.com/tendermint/go-amino v0.16.0 // indirect
+	github.com/zondax/hid v0.9.0 // indirect
+	go.etcd.io/bbolt v1.3.6 // indirect
+	go.opencensus.io v0.23.0 // indirect
+	golang.org/x/crypto v0.0.0-20220722155217-630584e8d5aa // indirect
+	golang.org/x/net v0.0.0-20220726230323-06994584191e // indirect
+	golang.org/x/sys v0.0.0-20220811171246-fbc7d0a398ab // indirect
+	golang.org/x/term v0.0.0-20220722155259-a9ba230a4035 // indirect
+	golang.org/x/text v0.3.7 // indirect
+	google.golang.org/protobuf v1.28.1 // indirect
+	gopkg.in/check.v1 v1.0.0-20201130134442-10cb98267c6c // indirect
+	gopkg.in/ini.v1 v1.66.6 // indirect
+	gopkg.in/yaml.v3 v3.0.1 // indirect
+	nhooyr.io/websocket v1.8.6 // indirect
+)
+
+replace (
+	// branch: v0.27.0.rc3-osmo, current tag: v0.27.0.rc3-osmo
+	github.com/CosmWasm/wasmd => github.com/osmosis-labs/wasmd v0.27.0-rc2.0.20220517191021-59051aa18d58
+	// Our cosmos-sdk branch is:  https://github.com/osmosis-labs/cosmos-sdk, current branch: osmosis-main. Direct commit link: https://github.com/osmosis-labs/cosmos-sdk/commit/5c9a51c277d067e0ec5cf48df30a85fae95bcd14
+	github.com/cosmos/cosmos-sdk => github.com/osmosis-labs/cosmos-sdk v0.45.1-0.20220822135534-5c9a51c277d0
+	// Use Osmosis fast iavl
+	github.com/cosmos/iavl => github.com/osmosis-labs/iavl v0.17.3-osmo-v7
+	// use cosmos-compatible protobufs
+	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
+	// use grpc compatible with cosmos protobufs
+	google.golang.org/grpc => google.golang.org/grpc v1.33.2
+)
```

Comment:
cc @puneet2019 will this cause import issues for you all? This comes from the `osmosis-labs/osmosis` import, but none of these replaces matter for you all.

The replaces won't happen when you import this module as only top-level package replaces get applied.

It may also be the case that osmosis-labs/osmosis/v11 doesn't get compiled, since its only imported in tests

Reply:
I will be trying this today or tomorrow! 
I had tried something similar but had import cycle issues.

Reply:
Go mod for the epochs module should be having an entry for the osmosis module itself, and that would need to be replaced with the file
Something like
```
require (
    "github.com/osmosis-labs/osmosis/v11" v11.0.1
    ...
)

replace (
    github.com/osmosis-labs/osmosis/v11 => ../../
)
```
This is something similar to what is being done on the cosmos-sdk, which basically took inspiration from k8s itself.

Reply:
yes. Better if we use this replace.

Reply:
Hey!
Could you possibly show an example on how to import this into other apps?
Here is what i tried.
- cloned an interchain-accounts repo
- cloned osmosis and switch to `epochs-module` branch.
- it won't let me `go get`
``` go get github.com/osmosis-labs/osmosis/x/epochs@epochs-module
go: github.com/osmosis-labs/osmosis/x/epochs@v0.0.0: reading github.com/osmosis-labs/osmosis/x/epochs/x/epochs/go.mod at revision x/epochs/v0.0.0: unknown revision x/epochs/v0.0.0
```

- set the go.mod as such

go.mod
```
module github.com/cosmos/interchain-accounts

go 1.18

replace github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1

replace (
	github.com/osmosis-labs/osmosis/x/epochs => ../osmosis/x/epochs
)
exclude (
	github.com/osmosis-labs/osmosis/v11 v11.0.1
)
require (
	github.com/osmosis-labs/osmosis/x/epochs v0.0.0
	github.com/cosmos/cosmos-sdk v0.45.6
	github.com/gogo/protobuf v1.3.3
	.........
```

- run `go mod tidy` 
```
➜  interchain-accounts git:(393d844) ✗ [31/08/22 4:44:58]$ go mod tidy
go: finding module for package github.com/osmosis-labs/osmosis/v11/app/apptesting
go: finding module for package github.com/osmosis-labs/osmosis/v11/osmoutils
go: finding module for package github.com/osmosis-labs/osmosis/v11/app
github.com/cosmos/interchain-accounts/app imports
        github.com/osmosis-labs/osmosis/x/epochs/types imports
        github.com/osmosis-labs/osmosis/v11/osmoutils: module github.com/osmosis-labs/osmosis@latest found (v1.0.4), but does not contain package github.com/osmosis-labs/osmosis/v11/osmoutils
github.com/cosmos/interchain-accounts/app imports
        github.com/osmosis-labs/osmosis/x/epochs/keeper tested by
        github.com/osmosis-labs/osmosis/x/epochs/keeper.test imports
        github.com/osmosis-labs/osmosis/v11/app: module github.com/osmosis-labs/osmosis@latest found (v1.0.4), but does not contain package github.com/osmosis-labs/osmosis/v11/app
github.com/cosmos/interchain-accounts/app imports
        github.com/osmosis-labs/osmosis/x/epochs/keeper tested by
        github.com/osmosis-labs/osmosis/x/epochs/keeper.test imports
        github.com/osmosis-labs/osmosis/v11/app/apptesting: module github.com/osmosis-labs/osmosis@latest found (v1.0.4), but does not contain package github.com/osmosis-labs/osmosis/v11/app/apptesting

```

Reply:
@puneet2019 the import actually works. For the other repo, we dont do replace, but `go get` the package directly instead.
Ran, and it seems to work
```
go get  github.com/osmosis-labs/osmosis/x/epochs@87034b3ca46c8a26812afe24a275882c65113dd0
```
Creates an entry in go.mod
```
github.com/osmosis-labs/osmosis/x/epochs v0.0.0-20220831122550-87034b3ca46c
```

Reply:
@Anmol1696 were you able to get it right? importing it into another repository?
`go get` works, but import still seems to be an issue

Reply:
@puneet2019 seems problem from conflict with sdk45 and sdk46. I'm trying to fix it.